### PR TITLE
Error on umasks other then 022

### PR DESCRIPTION
**Pull request by rhatdan**

### CODE REVIEW

**Comment by rhatdan** [comment]
@cgwalters PTAL

**Review by javipolo — APPROVED** [review]

File: recipes/common/Makefile.common [hunk 1]
```diff
@@ -156,8 +157,14 @@ install-chrome:
 		rm $(CHROME_DOWNLOAD_PATH); \
 	fi;
 
+.PHONY: check-umask
+check-umask:
+	@test "$(UMASK)" = "0022" || \
+		(echo; echo -n "Error: umask other then 0022 can cause unexpected behavour! "; \
```
Comment (cgwalters) [inline recipes/common/Makefile.common]: - nit: "behavior"
- Also we might as well print the umask, like "error: umask $(UMASK) will cause unexpected behavior: use umask 022"

And um...we should probably find a command that will forcibly reset the git checkout based on an updated umask, but I'm having trouble finding a nice way to do that that's sane; a brutal `rm * -rf; git reset --hard HEAD` works but...not my favorite

Comment (javipolo) [reply recipes/common/Makefile.common]: We could invent something using `git ls-files --stage` that lists all the file permissions in the repo

Comment (javipolo) [reply recipes/common/Makefile.common]: This did the trick for me, but it does not account for files with space in its name

Check that permissions are wrong in the first place
```
/tmp$ umask
0077
/t/ai-lab-recipes$ find training/common/usr | xargs ls -ld
drwx------. 4 javipolo javipolo   80 may  6 10:45 training/common/usr
drwx------. 3 javipolo javipolo   60 may  6 10:45 training/common/usr/lib
drwx------. 2 javipolo javipolo   60 may  6 10:45 training/common/usr/libexec
-rwx------. 1 javipolo javipolo 1253 may  6 10:45 training/common/usr/libexec/bootc-generic-growpart
drwx------. 3 javipolo javipolo   60 may  6 10:45 training/common/usr/lib/systemd
drwx------. 3 javipolo javipolo   80 may  6 10:45 training/common/usr/lib/systemd/system
-rw-------. 1 javipolo javipolo  670 may  6 10:45 training/common/usr/lib/systemd/system/bootc-generic-growpart.service
drwx------. 2 javipolo javipolo   60 may  6 10:45 training/common/usr/lib/systemd/system/local-fs.target.wants
lrwxrwxrwx. 1 javipolo javipolo   33 may  6 10:45 training/common/usr/lib/systemd/system/local-fs.target.wants/bootc-generic-growpart.service -> ../bootc-generic-growpart.service
```

Fix file and directory permissions
```
/t/ai-lab-recipes$ git ls-files --stage | awk '{print $1,$4}' | grep -vE '^120000' | while read mode file; do chmod $((mode % 1000)) $file; done
/t/ai-lab-recipes$ find -type d -exec chmod 755 '{}' \;
/t/ai-lab-recipes$ find training/common/usr | xargs ls -ld
drwxr-xr-x. 4 javipolo javipolo   80 may  6 10:45 training/common/usr
drwxr-xr-x. 3 javipolo javipolo   60 may  6 10:45 training/common/usr/lib
drwxr-xr-x. 2 javipolo javipolo   60 may  6 10:45 training/common/usr/libexec
-rwxr-xr-x. 1 javipolo javipolo 1253 may  6 10:45 training/common/usr/libexec/bootc-generic-growpart
drwxr-xr-x. 3 javipolo javipolo   60 may  6 10:45 training/common/usr/lib/systemd
drwxr-xr-x. 3 javipolo javipolo   80 may  6 10:45 training/common/usr/lib/systemd/system
-rw-r--r--. 1 javipolo javipolo  670 may  6 10:45 training/common/usr/lib/systemd/system/bootc-generic-growpart.service
drwxr-xr-x. 2 javipolo javipolo   60 may  6 10:45 training/common/usr/lib/systemd/system/local-fs.target.wants
lrwxrwxrwx. 1 javipolo javipolo   33 may  6 10:45 training/common/usr/lib/systemd/system/local-fs.target.wants/bootc-generic-growpart.service -> ../bootc-generic-growpart.service
```

Comment (rhatdan) [reply recipes/common/Makefile.common]: I would do.

u=$(umask)
umask 022
rm -rf ai-lab-recipes
git checkout ...
umask $u

**Pull request merged**